feature/s3/transfermanager/api_op_PutObject.go

```diff
@@ -0,0 +1,1077 @@
+}
+
+func (o *PutObjectOutput) mapFromPutObjectOutput(out *s3.PutObjectOutput, bucket, key string) {
+	o.BucketKeyEnabled = aws.ToBool(out.BucketKeyEnabled)
+	o.ChecksumCRC32 = aws.ToString(out.ChecksumCRC32)
+	o.ChecksumCRC32C = aws.ToString(out.ChecksumCRC32C)
+	o.ChecksumSHA1 = aws.ToString(out.ChecksumSHA1)
+	o.ChecksumSHA256 = aws.ToString(out.ChecksumSHA256)
+	o.ETag = aws.ToString(out.ETag)
+	o.Expiration = aws.ToString(out.Expiration)
+	o.Bucket = bucket
+	o.Key = key
+	o.RequestCharged = RequestCharged(out.RequestCharged)
+	o.SSEKMSKeyID = aws.ToString(out.SSEKMSKeyId)
+	o.ServerSideEncryption = ServerSideEncryption(out.ServerSideEncryption)
+	o.VersionID = aws.ToString(out.VersionId)
+	o.ResultMetadata = out.ResultMetadata.Clone()
+}
+
+func (o *PutObjectOutput) mapFromCompleteMultipartUploadOutput(out *s3.CompleteMultipartUploadOutput, bucket, uploadID string, completedParts completedParts) {
+	o.UploadID = uploadID
+	o.CompletedParts = completedParts
+	o.BucketKeyEnabled = aws.ToBool(out.BucketKeyEnabled)
+	o.ChecksumCRC32 = aws.ToString(out.ChecksumCRC32)
+	o.ChecksumCRC32C = aws.ToString(out.ChecksumCRC32C)
+	o.ChecksumSHA1 = aws.ToString(out.ChecksumSHA1)
+	o.ChecksumSHA256 = aws.ToString(out.ChecksumSHA256)
+	o.ETag = aws.ToString(out.ETag)
+	o.Expiration = aws.ToString(out.Expiration)
+	o.Bucket = bucket
+	o.Key = aws.ToString(out.Key)
+	o.RequestCharged = RequestCharged(out.RequestCharged)
+	o.SSEKMSKeyID = aws.ToString(out.SSEKMSKeyId)
+	o.ServerSideEncryption = ServerSideEncryption(out.ServerSideEncryption)
+	o.VersionID = aws.ToString(out.VersionId)
+	o.ResultMetadata = out.ResultMetadata
+}
+
+// PutObject uploads an object to S3, intelligently buffering large
+// files into smaller chunks and sending them in parallel across multiple
+// goroutines. You can configure the chunk size and concurrency through the
+// Options parameters.
+//
+// Additional functional options can be provided to configure the individual
+// upload. These options are copies of the Uploader instance Upload is called from.
+// Modifying the options will not impact the original Uploader instance.
+//
+// It is safe to call this method concurrently across goroutines.
+func (c *Client) PutObject(ctx context.Context, input *PutObjectInput, opts ...func(*Options)) (*PutObjectOutput, error) {
+	i := uploader{in: input, options: c.options.Copy()}
+	for _, opt := range opts {
+		opt(&i.options)
+	}
+
+	return i.upload(ctx)
+}
+
+type uploader struct {
+	options Options
+
+	in *PutObjectInput
+
+	// PartPool allows for the re-usage of streaming payload part buffers between upload calls
+	partPool byteSlicePool
+}
+
+func (u *uploader) upload(ctx context.Context) (*PutObjectOutput, error) {
+	if err := u.init(); err != nil {
+		return nil, fmt.Errorf("unable to initialize upload: %w", err)
+	}
+	defer u.partPool.Close()
+
+	clientOptions := []func(o *s3.Options){
+		func(o *s3.Options) {
+			o.APIOptions = append(o.APIOptions,
```

Comment:
These options don't seem dynamic, can we add them when we create the client instead of when we call `upload` ?